Lib/weakref.py

```diff
@@ -308,9 +252,7 @@ def valuerefs(self):
         keep the values around longer than needed.
 
         """
-        if self._pending_removals:
-            self._commit_removals()
-        return list(self.data.values())
+        return list(self.data.copy().values())
```

Comment:
I think the copy is not necessary here, although it's not really a problem. This can be:

```python
return list(self.data.values())
```

In the default build, the GIL ensures the list construction from the the `dict_values` is atomic. In the free-threaded build, the critical section ensures the same thing.

Reply:
Okay, I didn't know that, as you said copying is not an issue so I am leaving that.